packages/integration-tests/src/package_manager_sanity_checks.test.ts

```diff
@@ -81,77 +81,142 @@ void describe('getting started happy path', async () => {
-      path.join('auth', 'resource.ts'),
-      'backend.ts',
-      path.join('data', 'resource.ts'),
-      'package.json',
-      'tsconfig.json',
-    ];
-
-    assert.deepStrictEqual(
-      files.sort(),
-      expectedAmplifyFiles.map((suffix) => path.join(pathPrefix, suffix))
-    );
+    const TIMEOUT_MS = 1000 * 60 * 3; // 3 minutes
+    const startTime = Date.now();
 
-    await runWithPackageManager(
-      packageManager,
-      [
-        'ampx',
-        'pipeline-deploy',
-        '--branch',
-        branchBackendIdentifier.name,
-        '--appId',
-        branchBackendIdentifier.namespace,
-      ],
-      tempDir,
-      { env: { CI: 'true' } }
-    ).run();
-
-    const clientConfigStats = await fsp.stat(
-      await getClientConfigPath(ClientConfigFileBaseName.DEFAULT, tempDir)
-    );
+    while (Date.now() - startTime < TIMEOUT_MS) {
+      try {
+        if (packageManager === 'yarn-classic') {
+          await execa('yarn', ['add', 'create-amplify'], { cwd: tempDir });
+          await execaCommand(
+            './node_modules/.bin/create-amplify --yes --debug',
+            {
+              cwd: tempDir,
+              env: { npm_config_user_agent: 'yarn/1.22.21' },
+            }
+          );
+        } else {
+          await runPackageManager(
+            packageManager,
+            ['create', amplifyAtTag, '--yes'],
+            tempDir
+          ).run();
+        }
+
+        const pathPrefix = path.join(tempDir, 'amplify');
+
+        const files = await glob(path.join(pathPrefix, '**', '*'), {
+          nodir: true,
+          windowsPathsNoEscape: true,
+          ignore: ['**/node_modules/**', '**/yarn.lock'],
+        });
+
+        const expectedAmplifyFiles = [
+          path.join('auth', 'resource.ts'),
+          'backend.ts',
+          path.join('data', 'resource.ts'),
+          'package.json',
+          'tsconfig.json',
+        ];
+
+        assert.deepStrictEqual(
+          files.sort(),
+          expectedAmplifyFiles.map((suffix) => path.join(pathPrefix, suffix))
+        );
+
+        await runWithPackageManager(
+          packageManager,
+          [
+            'ampx',
+            'pipeline-deploy',
+            '--branch',
+            branchBackendIdentifier.name,
+            '--appId',
+            branchBackendIdentifier.namespace,
+          ],
+          tempDir,
+          { env: { CI: 'true' } }
+        ).run();
+
+        const clientConfigStats = await fsp.stat(
+          await getClientConfigPath(ClientConfigFileBaseName.DEFAULT, tempDir)
+        );
+
+        assert.ok(clientConfigStats.isFile());
+
+        // If we reach here without errors, break the loop
+        return;
+      } catch (error) {
+        console.error('Error occurred:', error);
+
+        const errorMessage = error instanceof Error ? error.message : '';
 
-    assert.ok(clientConfigStats.isFile());
+        const isCommandFailedError = errorMessage.includes('exit code 1');
```

Comment:
```suggestion
        const isFailedInstallDepsError = errorMessage.includes('exit code 1') && (errorMessage.includes('@aws-amplify/backend') || errorMessage.includes('aws-amplify'));
```
I think we should check if error message has this and one of the dependencies we install during create amplify. `exit code 1` may show up in other errors so we should be more explicit here while keeping it package manager agnostic.